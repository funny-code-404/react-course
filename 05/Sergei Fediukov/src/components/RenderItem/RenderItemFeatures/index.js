const RenderItemFeatures = (props) => {
    const { features } = props
    return (
        <>
            {
                features.map(x => {
                    for (const key in x) {
                        return (
                            x[key] &&
                            <p key={key}>{key.charAt(0).toUpperCase() + key.substr(1).replace(/\_/g, ' ')}: <span>{x[key]}</span></p>
                        )
                    }
                })
            }
        </>
    )
}
export default RenderItemFeatures