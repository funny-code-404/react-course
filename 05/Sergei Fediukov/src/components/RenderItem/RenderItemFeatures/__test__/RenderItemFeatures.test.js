import { configure, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

import RenderItemFeatures from '..'

configure({ adapter: new Adapter() });

describe('Should render correctly', () => {
    const props = {
        features: [{ test: 'test' }],
    }
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<RenderItemFeatures {...props} />)
    })
    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1)
    })
})