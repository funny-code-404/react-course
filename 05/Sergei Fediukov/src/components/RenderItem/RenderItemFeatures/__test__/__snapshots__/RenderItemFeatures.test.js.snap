// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly to Match Snapshot 1`] = `
<Fragment>
  <p
    key="test"
  >
    Test
    : 
    <span>
      test
    </span>
  </p>
</Fragment>
`;
